### DISCUSSION TITLE
How to Update?

### DISCUSSION THREAD

**Discussion by ederpardeiro** [microsoft/winget-pkgs#174317] [General]
Hi guys!

Recently I create a pull request to be uptates my software using winget. Now I can list there with "winget search rmc".
Reading some documents in the internet, I understand that to update my software, I need to change Hash and version in the YAML file on the GitHub.
I change it and upload the new installer (1.8), but when I try the "winget search rmc" command, the older version (1.7) appear. See bellow:

![image](https://github.com/user-attachments/assets/cf216337-dff3-42e9-a50a-59b36c0e8496)

![image](https://github.com/user-attachments/assets/d4c3bd4e-65fd-4de2-bce4-362b3686aedb)

How could I to fix it?

Sorry, but I do not speak fluently english

**Comment by Trenly** [comment]
I only see a PR for the 1.7 version - https://github.com/microsoft/winget-pkgs/pulls?q=is%3Apr+author%3Aederpardeiro

For each new version, the PR will need to be created updating the YAML

**Reply by ederpardeiro** [reply]
Ok, I changed the name of this PR from "New package: ConvectivaTecnologia.RMCConvectiva version 1.7" to "ConvectivaTecnologia.RMCConvectiva". Now I don't do reference for version in PR name.
I really don't undertand how to do it... this is my first project here... 
I need to create a new PR? Isn't sufficiente change informations in the YAML file?

**Reply by Trenly** [reply]
WinGet only operates out of https://github.com/microsoft/winget-pkgs/tree/master 

In order for updates to be shown in the command line -
1) Make a new branch in your fork
2) Update the YAML in the branch
3) Make pull request for that branch into microsoft/winget-pkgs@main
4) Wait for PR to merge
5) Wait for https://github.com/microsoft/winget-pkgs/labels/Publish-Pipeline-Succeeded label 

Example - https://github.com/microsoft/winget-pkgs/pull/173966/files

**Comment by ederpardeiro** [comment]
Sorry, but I have no idea how to do this. This is my first and single projetc. I’m not developer.

 

The only thing I need is to be able to update my software easily through Winget and I also need to be able to, when I make a modification, make it available for update.

 

I will read more about it, after that, When I can understand, I will try again. 

 

Could you recommended some article, documentation or vídeo where I can learn about it?

 

Sds.,

  _____  

	

 

 

 

De: Kaleb Luedtke ***@***.***> 
Enviada em: quinta-feira, 19 de setembro de 2024 13:54
Para: microsoft/winget-pkgs ***@***.***>
Cc: Eder Pardeiro ***@***.***>; Author ***@***.***>
Assunto: Re: [microsoft/winget-pkgs] How to Update? (Discussion #174317)

 

WinGet only operates out of https://github.com/microsoft/winget-pkgs/tree/master

In order for updates to be shown in the command line -

1.	Make a new branch in your fork
2.	Update the YAML in the branch
3.	Make pull request for that branch into ***@***.***
4.	Wait for PR to merge
5.	Wait for Publish-Pipeline-Succeeded <https://github.com/microsoft/winget-pkgs/labels/Publish-Pipeline-Succeeded>  label

Example - https://github.com/microsoft/winget-pkgs/pull/173966/files

**Reply by Trenly** [reply]
https://learn.microsoft.com/en-us/shows/open-at-microsoft/wingetcreate-keeping-winget-packages-up-to-date/
https://learn.microsoft.com/en-us/shows/open-at-microsoft/contributing-to-windows-package-manager/

**Reply by ederpardeiro** [reply]
Thank you very much. With this videos I was able to update the software and now they appear.

![image](https://github.com/user-attachments/assets/c228eb9c-3bf2-4171-9b87-679fff8f6f9c)

Anyway, I am still having a problem. When I run the command "winget search rmc", the older version appearing:

![image](https://github.com/user-attachments/assets/b7cbe17e-2bc5-45b7-96a9-9f4bd6b65efd)

I think this issue occur for this, but I don't know how to fix it:

![image](https://github.com/user-attachments/assets/9e1df3bd-4680-41ee-a82f-39e7593f0f02)
